src/storage/disk_manager.c

```diff
@@ -1404,15 +1432,6 @@ disk_rv_redo_format (THREAD_ENTRY * thread_p, const LOG_RCV * rcv)
 
       disk_Cache->perm_purpose_info.extend_info.nsect_total += volheader->nsect_total;
       disk_Cache->perm_purpose_info.extend_info.nsect_max += volheader->nsect_max;
-
-      if (is_tran_server_with_remote_storage ())
-	{
-	  // transaction servers with remote storage:
-	  //  - maintain a separate count of permanent volumes
-	  //  - have their own set of temporary volumes which they maintain using the disk Cache structures
-	  assert ((disk_Page_server_perm_volume_count + 1) == disk_Cache->nvols_perm);
-	  disk_Page_server_perm_volume_count++;
```

Comment:
In the undo (and CLR) function of RVDK_FORMAT, the disk_Page_server_perm_volume_count is decreased when it's rollbacked. I'm concerned that the corresponding operations are done in different recovery indexes' redo and undo, increasing in REDO of RVDK_NEWVOL and decreasing in UNDO of RVDK_FORMAT.

Reply:
In disk_format(), there is a window in which the UNDO log of RVDK_FORMAT is appended, but the REDO log of TVDK_NEWVOL is not appended. If a crash or an error occurs in the windows, it seems to be a problem.
 By the way, similarly, there is already a window in which UNDO log of RVDK_FORAT is appended, but the REDO log of RVDK_FORMAT is not appended. I think it already has the same problem.

Reply:
OK, I agree. Thanks for detail feedback. 

In case of 
```
RVDK_FORMAT (undo)
RVDK_NEWVOL (redo)

crash!

RVDK_FORMAT (redo)
RVDK_FORMAT (redo)
```
disk_Page_server_perm_count is increased by `RVDK_NEWVOL`  redo log, 
but it can not be decreased by `RVDK_FORMAT` undo log (CLR). 
Because disk volume header is not loaded on PTS yet.. 

Maybe this should be fixed by other way, 
such as
1. changing the fetch mode  from OLD_PAGE_IF_IN_BUFFER_OR_IN_TRANSIT to OLD_PAGE while replicating this kind of log record.
2. or comparing the volume id with disk_Page_server_perm_count when it needs to be increased or decreased .

Reply:
2 can not be a solution, because PTS keeps getting the wrong number of permanent volumes on the page server and tries to fetch the page using it.